File: src/Notifications.Api/Entities/Notification.cs
using StarterKit.Notifications.Contracts.SystemNotifications;
using StarterKit.Shared.Entities;

namespace StarterKit.Notifications.Api.Entities;

public class Notification : AuditableEntity
{
    public string FromUserId { get; set; } = null!;

    public string? FromName { get; set; }

    public string ToUserId { get; set; } = null!;

    public string Title { get; set; } = null!;

    public string? Message { get; set; }

    public string? Url { get; set; }

    public NotificationStatus Status { get; set; }
}

File: clients/admin/src/features/notifications/types/notification.ts
/** Mirrors `Notifications.Contracts/SystemNotifications/NotificationState.cs` — the backend serializes this enum by name, not by number. */
export enum NotificationStatus {
  None = "None",
  Read = "Read",
  Archived = "Archived",
}

export interface NotificationDto {
  id: string;
  fromUserId: string;
  fromName?: string | null;
  toUserId: string;
  title: string;
  message?: string | null;
  url?: string | null;
  status: NotificationStatus;
  created: string;
}

export interface NotificationLookupParams {
  toUserId?: string;
  status?: NotificationStatus;
  pageNumber?: number;
  pageSize?: number;
}
